mm2src/coins/nft/nft_structs.rs

```diff
@@ -312,53 +362,55 @@ impl fmt::Display for TransferStatus {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct NftTransferHistory {
     pub(crate) chain: Chain,
     pub(crate) block_number: u64,
     pub(crate) block_timestamp: u64,
-    pub(crate) block_hash: String,
+    pub(crate) block_hash: Option<String>,
     /// Transaction hash in hexadecimal format
     pub(crate) transaction_hash: String,
-    pub(crate) transaction_index: u64,
-    pub(crate) log_index: u64,
-    pub(crate) value: BigDecimal,
+    pub(crate) transaction_index: Option<u64>,
+    pub(crate) log_index: Option<u64>,
+    pub(crate) value: Option<BigDecimal>,
     pub(crate) contract_type: ContractType,
-    pub(crate) transaction_type: String,
+    pub(crate) transaction_type: Option<String>,
     pub(crate) token_address: String,
     pub(crate) token_id: BigDecimal,
+    pub(crate) token_uri: Option<String>,
     pub(crate) collection_name: Option<String>,
-    pub(crate) image: Option<String>,
+    pub(crate) image_url: Option<String>,
     pub(crate) token_name: Option<String>,
     pub(crate) from_address: String,
     pub(crate) to_address: String,
     pub(crate) status: TransferStatus,
     pub(crate) amount: BigDecimal,
-    pub(crate) verified: u64,
+    pub(crate) verified: Option<u64>,
     pub(crate) operator: Option<String>,
-    pub(crate) possible_spam: Option<bool>,
+    pub(crate) possible_spam: bool,
 }
 
 #[derive(Debug, Deserialize)]
 pub(crate) struct NftTransferHistoryWrapper {
```

Comment:
Some fields which should be BigDecimal or u64 are just strings in moralis response, that's why I added `SerdeStringWrap` structure. In addition `contract_type` from moralis can be null, but in our structs it must not be null, its not Option field. Right now we easily can build raw structure from moralis response and change/filter it as we want.

well I can implement function like `from_value` for `Nft` and `TxHistory` structures in which we build them field by field from json value. But I think the ability to just use`serde_json::from_str` is more preferable.

Reply:
> Some fields which should be BigDecimal or u64 are just strings in moralis response

Do we have to deserialize and keep them in string format? You should be able handle this on deserialization. If the response from moralis doesn't fit in the struct, we can return an error.

Reply:
> > Some fields which should be BigDecimal or u64 are just strings in moralis response
> 
> Do we have to deserialize and keep them in string format? You should be able handle this on deserialization. If the response from moralis doesn't fit in the struct, we can return an error.

we also constantly deserialize and serialize structures for database. And I need to keep block number as u64. It is more comfortable to make block number and amount and other fields as types what we want from the beginning then use `from_string` every time when we need this fields.

> If the response from moralis doesn't fit in the struct, we can return an error.

No if contact type is null we skip that structure
```rust
let contract_type = match nft_wrapper.contract_type {
                    Some(contract_type) => contract_type.0,
                    None => continue,
                };
```

Reply:
I don't get the point of not being able to deserialize the fields in the proper format instead of deserializing one value for string, null and u64/BigDecimal. But I guess it's fine then.

Reply:
> I don't get the point of not being able to deserialize the fields in the proper format instead of deserializing one value for string, null and u64/BigDecimal. But I guess it's fine then.

Im trying to say that fields from moralis dont have necessary types which we want and some of then can be null, so we need handle it. if you want to have one common structure, then you should impl some function which parses structure from `moralis` json so you will have fields with the right types from the begging. instead of this I just created wrappers and use serde_json::from_str

Reply:
> which parses structure from moralis json so you will have fields with the right types from the begging

Yeah this is what I was trying to suggest, we already doing this in many parts of mm2 source. Duplicating structs doesn't seem good solution to me.

Reply:
> > which parses structure from moralis json so you will have fields with the right types from the begging
> 
> Yeah this is what I was trying to suggest, we already doing this in many parts of mm2 source. Duplicating structs doesn't seem good solution to me.

https://github.com/KomodoPlatform/atomicDEX-API/commit/03e1a153936eef9d36c9743071558d91d688f86e
Personally I dont like this way of handling fields from moralis response. 
Instead of just using `NftWrapper` structure and building it with `serde_json::from_str` function I have to process each field separately. You  move the logic of collecting the structure o your custom `from_value` function and complicate your work.
Wrapper structures in nft carry a certain logic, they collect initial structure from the moralis response and already have u64 and BigDecimal fields.

@ozkanonur @shamardy

Reply:
But this isn't a good/better solution either. This approach basically makes the binary size 2x. You can at least unify the shared fields if any other way is not possible.

Reply:
@ozkanonur Hey
I created `NftCommon` and `NftTransferCommon` structures which contain common fields for structures and also renamed nft wrap structures to `..FromMoralis` and used `_moralis` names for variables in nft.rs

Reply:
Yeah this looks much better, thanks!